### PR TITLE
Add validation to rule data

### PR DESCRIPTION
**Pull request by lcarva**
This commit enhances the validation performed on rule data. It defines a JSON Schema for each value and provides a policy rules to execute them.

Ref: EC-132

### CODE REVIEW

File: policy/release/collection/rule-data-validation.rego [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#
+# METADATA
+# title: rule-data-validation
```
Comment (zregvart) [inline policy/release/collection/rule-data-validation.rego:3]: Perhaps something shorter like:

```suggestion
# title: policy-data
```

File: policy/beta/packages.rego [hunk 2]
```diff
@@ -66,3 +84,60 @@ _matches_version(version, matcher) if {
 } else := false
 
 _to_semver(v) := trim_prefix(v, "v")
+
+# Verify disallowed_packages is a non-empty list of objects
```
Comment (zregvart) [inline policy/beta/packages.rego:88]: The list is allowed to be empty
```suggestion
# Verify disallowed_packages is an array of objects with expected properties
```

File: policy/beta/packages.rego [hunk 3]
```diff
@@ -66,3 +84,60 @@ _matches_version(version, matcher) if {
...
+	pkg.format in {"semver", "semverv"}
+	some attr in ["min", "max"]
+
+	version := _to_semver(object.get(pkg, attr, ""))
+	version != ""
+
+	not semver.is_valid(version)
+
+	msg := sprintf(
+		"Item at index %d in %s does not have a valid %s semver value: %s",
```
Comment (zregvart) [inline policy/beta/packages.rego:138]: Perhaps we should quote the user provided value

```suggestion
		"Item at index %d in %s does not have a valid %s semver value: %q",
```

File: policy/beta/packages.rego (hunk 2 above)
Comment (lcarva) [reply policy/beta/packages.rego]: Thanks!

File: policy/release/collection/rule-data-validation.rego (hunk 1 above)
Comment (lcarva) [reply policy/release/collection/rule-data-validation.rego]: Decided to go with `policy_data` since that's the case rego usually expects.

**Review by zregvart — APPROVED** [review]
:shipit:

**Pull request merged**